sqlx-core/src/logger.rs

```diff
@@ -122,37 +157,34 @@ impl<'q> QueryLogger<'q> {
                     String::new()
                 };
 
-                if was_slow {
+                self.span.record("summary", summary);
+                self.span.record("db.statement", sql);
+                self.span.record("rows_affected", self.rows_affected);
+                self.span.record("rows_returned", self.rows_returned);
+
+                let _e = self.span.enter();
+                if was_slow { 
                     private_tracing_dynamic_event!(
                         target: "sqlx::query",
                         tracing_level,
-                        summary,
-                        db.statement = sql,
-                        rows_affected = self.rows_affected,
-                        rows_returned = self.rows_returned,
-                        // Human-friendly - includes units (usually ms). Also kept for backward compatibility
-                        ?elapsed,
-                        // Search friendly - numeric
-                        elapsed_secs = elapsed.as_secs_f64(),
-                        // When logging to JSON, one can trigger alerts from the presence of this field.
                         slow_threshold=?self.settings.slow_statements_duration,
+                         // Human-friendly - includes units (usually ms). Also kept for backward compatibility
+                         ?elapsed,
+                         // Search friendly - numeric
+                         elapsed_secs = elapsed.as_secs_f64(),
```

Comment:
Indentation looks like it might trip up the format linting, and in any case for git history it would be nice to avoid reordering the fields.

Reply:
Indentation addressed (sorry, the formatter is still not good at catching things in macro invocations). Can you clarify the issue with field ordering? The shared fields were moved into the span, but in the same order

Reply:
Unfortunately the fields need to also be kept in the existing logging after being added to the new span.

Moving them from the individual log statement to the span changes the json structure that loggers like `tracing-subscriber` with the `json` feature will output - with that crate the items move from `.fields` to `.span` - which is good for standardization, but breaks existing consumers. We need both to avoid a breaking change.

Reply:
I see, there doesn't currently seem to be a way to configure the json formatter to flatten the span fields into the event output (though maybe there should be?).

Would it be acceptable to gate some of this behavior behind a feature flag? It sounds like the span isn't really necessary for the slow query logger, and the event isn't really necessary for opentelemetry tracing. It would also be nice for those two things to work well together with a minimum of duplication.

Reply:
The slow query logger can already be configured to never trigger.

For example, if span tracking was added, but no existing logging was diminished or removed, a crate user could set `slow_statements_level` to `LevelFilter::Off` and rely exclusively on metrics reported on the span for their monitoring and alerting.

Reply:
I suppose that's true. I'm reusing this config to control span visibility as well, so I'll rework how that happens.